### PR TITLE
feat: gstreamer removed

### PR DESCRIPTION
**Pull request by klepo**
Gstreamer is no longer used - big changes
H264 (pyav) used
Better named logger
Examples renamed (no http and gstreamer)

### CODE REVIEW

File: era_5g_interface/h264_decoder.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    def __init__(self, fps: float, width: int, height: int) -> None:
+        self.decoder: VideoCodecContext = CodecContext.create("h264", "r")
+        self.decoder.width = width
+        self.decoder.height = height
+        self.decoder.framerate = fps
+        self.decoder.pix_fmt = "yuv420p"
+
+    def decode_packet_data(self, packet_data, format: str = "bgr24"):
+        packet = Packet(packet_data)
+        # try:
```
Comment (ZdenekM) [inline era_5g_interface/h264_decoder.py:23]: Remove commented code or add an explanation of why it is commented (if it is important stuff).

File: requirements.txt [hunk 2]
```diff
@@ -1 +1,4 @@
-opencv-python~=4.6.0.37  # for usage with GStreamer, one has to compile OpenCV with its support turned on
+opencv-python
+av
```
Comment (ZdenekM) [inline requirements.txt:2]: There should be specific versions.

File: era_5g_interface/h264_encoder.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class H264Encoder:
+    def __init__(self, fps: float, width: int, height: int) -> None:
+        self.encoder: VideoCodecContext = CodecContext.create("h264", "w")
+        self.encoder.width = width
+        self.encoder.height = height
+        self.encoder.framerate = fps
+        self.encoder.pix_fmt = "yuv420p"
+        # self.encoder.options = {"crf": "0", "preset": "ultrafast", "tune": "zerolatency"}
```
Comment (ZdenekM) [inline era_5g_interface/h264_encoder.py:21]: Remove, make it an option, or document why it is commented if it makes sense to keep it here.

File: era_5g_interface/interface_helpers.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+MAX_LATENCY = float(os.getenv("NETAPP_MAX_LATENCY", 100))
+
+
+# Class for holding data about processing times (latency)
+class LatencyMeasurements:
+    def __init__(self, num_latencies_to_keep: int = 10) -> None:
+        self.num_latencies_to_keep = num_latencies_to_keep
+        self.processing_latencies = np.zeros(num_latencies_to_keep)
+
+    def store_latency(self, latency) -> None:
```
Comment (ZdenekM) [inline era_5g_interface/interface_helpers.py:24]: Add type annotation for `latency`.

File: era_5g_interface/interface_helpers.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        # Remove the oldest entry and add the new one
+        # Using strategy "Copy one before and substitute at the end" (fastest)
+        self.processing_latencies[0:-1] = self.processing_latencies[1:]
+        self.processing_latencies[-1] = latency
+
+    def get_latencies(self):
```
Comment (ZdenekM) [inline era_5g_interface/interface_helpers.py:30]: Add type annotation.

File: era_5g_interface/interface_helpers.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        self.session.mount(MIDDLEWARE_ADDRESS, self.adapter)
+        self.connection_error = False
+
+    def send_middleware_heart_beat_request(self, headers: dict, json: dict) -> None:
+        if not self.connection_error:
+            logger.info(f"Sending heart beat to middleware: {json}")
+            try:
+                response = self.session.post(MIDDLEWARE_ADDRESS, headers=headers, json=json, timeout=(0.2, 0.2))
+                logger.info(f"Middleware heart_beat response: {response.text}")
+            except Exception as e:
```
Comment (ZdenekM) [inline era_5g_interface/interface_helpers.py:52]: Use some more specific exception, e.g., `HTTPError`.

File: era_5g_interface/interface_helpers.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+        data = {
+            "Id": NETAPP_ID,
+            "CurrentRobotsCount": current_robot_count,
+            "OptimalLimit": optimal_robot_count_limit,
+            "HardLimit": hard_robot_count_limit,
+        }
+        headers = {"Content-type": "application/json"}
+        self.send_middleware_heart_beat_request(headers=headers, json=data)
+        # logger.info(f"Heart beat published to middleware: {data}")
```
Comment (ZdenekM) [inline era_5g_interface/interface_helpers.py:86]: remove

**Review by ZdenekM — CHANGES_REQUESTED** [review]

File: era_5g_interface/h264_decoder.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        # TODO multiple frames?
+        # arrays = []
+        # logger.info(f"Decoding packet: {packet}")
+        for frame in self.decoder.decode(packet):
```
[the remark this answers is not in the document]
Comment (klepo) [reply era_5g_interface/h264_decoder.py:27]: yes

File: era_5g_interface/interface_helpers.py (hunk 5 above)
Comment (klepo) [reply era_5g_interface/interface_helpers.py:30]: Any

**Pull request closed without merging**